### PR TITLE
Udp KLV Decoding

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Adds a step the the UDP stream processing to make stream KLV data available. 
Also added some additional properties to the configuration to help with stream associations

### CODE REVIEW

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
build now

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/klv/KlvStreamService.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.alliance.video.stream.mpegts.klv;
+
+import java.util.Collection;
+import java.util.Map;
+
+public interface KlvStreamService {
+  /**
+   * Get the current kvl data mappings associated with the given parent id.
```
Comment (sgalla) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/klv/KlvStreamService.java:21]:
```suggestion
   * Get the current klv data mappings associated with the given parent id.
```

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/klv/KlvStreamService.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+   *
+   * @param key the klv properties key
+   * @param value the klv property value to filter on
+   * @return The current klv data the streams with matching key/value. Will return an empty
+   *     collection if no klv has been reported recently for the parameters.
+   */
+  Collection<KlvData> getStreamsKlvByParam(String key, String value);
+
+  /**
+   * Get the current kvl data mappings organized by parent stream id.
```
Comment (sgalla) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/klv/KlvStreamService.java:31]:
```suggestion
   * Get the current klv data mappings organized by parent stream id.
```

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import org.codice.alliance.libs.stanag4609.DecodedKLVMetadataPacket;
+import org.codice.alliance.libs.stanag4609.PESUtilities;
+import org.codice.alliance.video.stream.mpegts.Context;
+import org.codice.alliance.video.stream.mpegts.klv.KlvConsumer;
+import org.codice.alliance.video.stream.mpegts.klv.KlvData;
+import org.codice.ddf.libs.klv.KlvContext;
+import org.codice.ddf.libs.klv.KlvDataElement;
+import org.codice.ddf.libs.klv.KlvDecoder;
+
+class PESPacketToKVLPacketDecoder extends MessageToMessageDecoder<PESPacket> {
```
Comment (sgalla) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java:37]:
```suggestion
class PESPacketToKLVPacketDecoder extends MessageToMessageDecoder<PESPacket> {
```

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+  private final KlvDecoder decoder;
+
+  private final KlvConsumer klvService;
+
+  private final Context context;
+
+  private Map<String, String> properties = new HashMap<>();
+
+  public PESPacketToKVLPacketDecoder(
```
Comment (sgalla) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java:47]:
```suggestion
  public PESPacketToKLVPacketDecoder(
```

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    this.decoder = klvDecoder;
+    this.klvService = klvService;
+    this.context = context;
+    properties.putAll(context.getUdpStreamProcessor().getAdditionalProperties());
+    context
+        .getParentMetacard()
+        .ifPresent(metacard -> properties.put(PARENT_ID_KEY, metacard.getId()));
+  }
+
+  public PESPacketToKVLPacketDecoder(KlvConsumer service, Context context) {
```
Comment (sgalla) [inline catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java:58]:
```suggestion
  public PESPacketToKLVPacketDecoder(KlvConsumer service, Context context) {
```

File: catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoderTest.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  private KlvConsumer consumer;
+
+  private MetacardImpl parent;
+
+  private UdpStreamProcessor processor;
+
+  private PESPacketToKVLPacketDecoder decoder;
+
+  private Map<String, String> properties = new HashMap<>();
+  // getClass().getResourceAsStream("/test.jpg"),
```
Comment (sgalla) [inline catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoderTest.java:50]: extra comment?

**Comment by clockard** [comment]
build now

File: catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoderTest.java (hunk 6 above)
Comment (clockard) [reply catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoderTest.java]: Removed

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/klv/KlvStreamService.java (hunk 2 above)
Comment (clockard) [reply catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/klv/KlvStreamService.java]: Fixed

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java (hunk 3 above)
Comment (clockard) [reply catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java]: Fixed

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java (hunk 4 above)
Comment (clockard) [reply catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java]: Fixed

File: catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java (hunk 5 above)
Comment (clockard) [reply catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/PESPacketToKVLPacketDecoder.java]: Fixed

**Pull request merged**